### PR TITLE
[DOCS] Edits create data frame analytics job API

### PR DESCRIPTION
**Pull request by lcawl**
This PR updates the create data frame analytics job API (https://www.elastic.co/guide/en/elasticsearch/reference/master/put-dfanalytics.html) related to the addition of multiclass classification. For example, it updates the dependent_variable and num_top_classes definitions.

It also edits some existing unrelated content. For example, it comments out hyperparameter optimization details that are moving in https://github.com/elastic/stack-docs/pull/990 and it moves details about mappings from the create DFA job API to the start DFA job API.

Preview:
* http://elasticsearch_54751.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/put-dfanalytics.html
* http://elasticsearch_54751.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/start-dfanalytics.html

### CODE REVIEW

File: docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc [hunk 1]
```diff
@@ -46,19 +46,17 @@ instead of the primary headers.
...
 `index.number_of_shards` and `index.number_of_replicas` settings of the source 
-index will be copied over the destination index. When the source index matches 
-multiple indices, these settings will be set to the maximum values found in the 
-source indices.
+index are copied over the destination index. When the source index matches 
+multiple indices, these settings are set to the maximum values found in the 
+source indices. The mappings of the source indices are also copied over to the
+destination index. If the mappings of any of the fields don't match the source
+indices, however, the attempt fails with an error message.
+//TBD Does this cause the job creation to fail? What are the implications?
```
Comment (dimitris-athanasiou) [inline docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc:56]: We copy the mappings to the dest index when the `_start` API is called. Indeed, if the mappings are conflicted, the start API fails with an error. But during job creation we don't check.

File: docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc [hunk 2]
```diff
@@ -27,57 +27,46 @@ built-in roles and privileges:
...
+indices and stores the outcome in a destination index.
+
+If the destination index does not exist, it is created automatically when you
+start the job. See <<start-dfanalytics>>.
+
+If you supply only a subset of the {regression} or {classification} parameters,
+_hyperparameter optimization_ occurs. It determines a value for each of the
+undefined parameters. A fixed number of rounds is used for optimization which
+depends on the number of parameters being optimized.
+// TBD: Does this "fixed number of rounds" sentence mean that the more undefined
```
Comment (dimitris-athanasiou) [inline docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc:53]: @tveasey Could you help us clarify this please?

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

**Comment by dimitris-athanasiou** [comment]
I LGTM'd too soon, didn't notice the `WIP` label

File: docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc (hunk 2 above)
Comment (lcawl) [reply docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc:53]: I will pursue this in https://github.com/elastic/stack-docs/pull/990 instead

File: docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc [hunk 3]
```diff
@@ -111,24 +105,29 @@ The configuration information necessary to perform
...
+any class. Defaults to `maximize_minimum_recall`.
 
 `dependent_variable`::::
 (Required, string)
 +
 include::{docdir}/ml/ml-shared.asciidoc[tag=dependent-variable]
 +
 The data type of the field must be numeric (`integer`, `short`, `long`, `byte`), 
-categorical (`ip` or `keyword`), or boolean.
+categorical (`ip` or `keyword`), or boolean. There must be be no more than 30
```
Comment (dimitris-athanasiou) [inline docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc:129]: typo: double `be`

**Review by dimitris-athanasiou — COMMENTED** [review]
Looks good! Just a small typo.

File: docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc (hunk 3 above)
Comment (lcawl) [reply docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc:129]: Good catch, thanks!

**Pull request merged**